cluster/cluster.go

```diff
@@ -15,4 +38,85 @@ func Init(name, version string, primary bool, started time.Time) {
 		primaryChange: time.Now(),
 		stateChange:   time.Now(),
 	}
+	Mode = m
+	go probePeers(interval)
+}
+
+func Stop() {
+	close(shutdown)
+}
+
+func AddPeer(remoteAddr *url.URL) {
+	mu.Lock()
+	peer := &Node{
+		remoteAddr:  remoteAddr,
+		stateChange: time.Now(),
+	}
+	peers = append(peers, peer)
+	go peer.Probe()
+	mu.Unlock()
+}
+
+func GetPeers() []*Node {
+	mu.Lock()
+	p := make([]*Node, len(peers))
+	copy(p, peers)
+	mu.Unlock()
+	return p
+}
+
+func probePeers(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	for {
+		select {
+		case <-shutdown:
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			mu.Lock()
+			for _, peer := range peers {
+				go peer.Probe()
+			}
+			mu.Unlock()
+		}
+	}
+}
+
+// return the list of peers to broadcast requests to
+// Only 1 peer per partition is returned
+func PeersForQuery() []*Node {
```

Comment:
I originally had PeersForQuery exclude ThisNode, but i then added it back in to allow future features that provide better scheduling.  

For example in future, we way want to extend the "state" of a node to differentiate between ready to handle queries and ready to serve data (ie due to just having started up and not having data in memory).   We could then preference other nodes that have been online longer over a node that has just started up.